m2cgen/ast.py

```diff
@@ -235,7 +246,7 @@ def __str__(self):
     (PowExpr, lambda e: [e.base_expr, e.exp_expr]),
     (VectorVal, lambda e: e.exprs),
     (IfExpr, lambda e: [e.test, e.body, e.orelse]),
-    ((ExpExpr, SqrtExpr, TanhExpr), lambda e: [e.expr]),
+    ((IdExpr, ExpExpr, SqrtExpr, TanhExpr), lambda e: [e.expr]),
```

Comment:
Not sure here: should it be completely transparent and shouldn't be counted here?..

To be honest, I'd better delegate this part to you @izeigerman because to my shame I still cannot say that I clearly understand how `SubroutinesMixin` works 🙁 .

Reply:
> I still cannot say that I clearly understand how SubroutinesMixin works

No worries, neither do I 😄 Unfortunately this part of the project is far from being pretty and it's basically an obscure heuristics. I hope I'll have better ideas in future or someone smarter than I will suggest better approaches. 

> should it be completely transparent and shouldn't be counted here?..
I'd say It should be completely transparent and excluded explicitly here https://github.com/BayesWitnesses/m2cgen/blob/9e5af793e8b389d0c55469df938e8ac3b3ccf90d/m2cgen/ast.py#L258

Reply:
On a second thought I realized that the function is called `count_exprs` so silently ignoring an expression is not the best practice. I'd say we should do one of the following:
1. Do nothing. I don't anticipate this expression to contribute into the count significantly. At least right now.
2. Explicitly exclude using the `exclude_list` argument in places where this function is used.

Reply:
I'm out! Please feel free to choose any variant you think fits here best in your vision of conceptual.

Reply:
I'd go with the option 2 then - explicitly excluding it in places of invocation.